Help please @jepetty @brianaeng: What is causing the pipeline to fail now❔ It appears the agents may be missing a component but that couldn't be the case (because other richnav pipelines build using the same agents).

``` text
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
```

--- a/.azure/pipelines/ci.yml
+++ b/.azure/pipelines/ci.yml
@@ -175,7 +175,7 @@ stages:
                 $(_BuildArgs)
                 $(_InternalRuntimeDownloadArgs)
                 $(Windows64LogArgs)
-        displayName: Build x64/arm64
+        displayName: Build x64
 
       # Build the x86 shared framework
       # This is going to actually build x86 native assets.

--- a/.azure/pipelines/richnav.yml
+++ b/.azure/pipelines/richnav.yml
@@ -6,10 +6,11 @@
 trigger:
   branches:
     include:
-    - blazor-wasm
     - main
     - release/*
-    - internal/release/*
+
+# Do not run this pipeline for PR validation.
+pr: none
 
 variables:
 - name: _BuildArgs

--- a/.azure/pipelines/richnav.yml
+++ b/.azure/pipelines/richnav.yml
@@ -14,32 +14,32 @@ trigger:
 variables:
 - name: _BuildArgs
   value: '/p:SkipTestBuild=true'
-- name: Windows86LogArgs
+- name: WindowsNonX64LogArgs
   value: -ExcludeCIBinaryLog
 
 stages:
 - stage: build
   displayName: Build
   jobs:
-  # Build Windows (x64/x86)
+  # Build Windows (x64/x86/arm64)
   - template: jobs/default-build.yml
     parameters:
       jobName: Windows_build
-      jobDisplayName: "Build: Windows x64/x86"
+      jobDisplayName: "Build: Windows x64/x86/arm64"
       enableRichCodeNavigation: true
       agentOs: Windows
       steps:
-      - script: ./build.cmd
+      - script: ./eng/build.cmd
                 -ci
-                -all
                 -arch x64
+                -all
                 /p:EnableRichCodeNavigation=true
                 $(_BuildArgs)
         displayName: Build x64
 
       # Build the x86 shared framework
       # This is going to actually build x86 native assets.
-      - script: ./build.cmd
+      - script: ./eng/build.cmd
                 -ci
                 -noBuildRepoTasks
                 -arch x86
@@ -47,17 +47,31 @@ stages:
                 -noBuildJava
                 -noBuildNative
                 /p:EnableRichCodeNavigation=true
+                /p:OnlyPackPlatformSpecificPackages=true
                 $(_BuildArgs)
-                $(Windows86LogArgs)
+                $(WindowsNonX64LogArgs)
         displayName: Build x86
 
-      # Windows installers bundle both x86 and x64 assets
-      - script: ./build.cmd
+      # Build the arm64 shared framework
+      - script: ./eng/build.cmd
+                -ci
+                -noBuildRepoTasks
+                -arch arm64
+                -noBuildJava
+                -noBuildNative
+                /p:EnableRichCodeNavigation=true
+                /p:OnlyPackPlatformSpecificPackages=true
+                $(_BuildArgs)
+                $(WindowsNonX64LogArgs)
+        displayName: Build ARM64
+
+      # Windows installers bundle x86/x64/arm64 assets
+      - script: ./eng/build.cmd
                 -ci
                 -noBuildRepoTasks
                 -buildInstallers
                 -noBuildNative
-                /p:AssetManifestFileName=aspnetcore-win-x64-x86.xml
                 /p:EnableRichCodeNavigation=true
                 $(_BuildArgs)
+                $(WindowsNonX64LogArgs)
         displayName: Build Installers

--- a/.azure/pipelines/richnav.yml
+++ b/.azure/pipelines/richnav.yml
@@ -33,7 +33,15 @@ stages:
       - script: ./eng/build.cmd
                 -ci
                 -arch x64
+                -buildNative
+                $(_BuildArgs)
+        displayName: Build x64 native assets
+      - script: ./eng/build.cmd
+                -ci
+                -arch x64
                 -all
+                -noBuildNative
+                -noBuildRepoTasks
                 /p:EnableRichCodeNavigation=true
                 $(_BuildArgs)
         displayName: Build x64
@@ -42,11 +50,11 @@ stages:
       # This is going to actually build x86 native assets.
       - script: ./eng/build.cmd
                 -ci
-                -noBuildRepoTasks
                 -arch x86
                 -all
                 -noBuildJava
                 -noBuildNative
+                -noBuildRepoTasks
                 /p:EnableRichCodeNavigation=true
                 /p:OnlyPackPlatformSpecificPackages=true
                 $(_BuildArgs)
@@ -56,10 +64,10 @@ stages:
       # Build the arm64 shared framework
       - script: ./eng/build.cmd
                 -ci
-                -noBuildRepoTasks
                 -arch arm64
                 -noBuildJava
                 -noBuildNative
+                -noBuildRepoTasks
                 /p:EnableRichCodeNavigation=true
                 /p:OnlyPackPlatformSpecificPackages=true
                 $(_BuildArgs)
@@ -69,9 +77,10 @@ stages:
       # Windows installers bundle x86/x64/arm64 assets
       - script: ./eng/build.cmd
                 -ci
-                -noBuildRepoTasks
                 -buildInstallers
+                -noBuildJava
                 -noBuildNative
+                -noBuildRepoTasks
                 /p:EnableRichCodeNavigation=true
                 $(_BuildArgs)
                 $(WindowsNonX64LogArgs)

